### PR TITLE
Add a Student Repertoire Table

### PR DESCRIPTION
**Pull request by nathan-lalli**
Adds a student repertoire table to allow for students to have more than one repertoire.

- Adds a studentrepertoire model for the table
- Adds the studentrepertoire to the index
- Adds a studentrepertoire controller
- Edits the controller for the pieces table
- Edits the controller for the repertoire table
- Adds foreign keys to the studentrepertoire table
- Edits the foreign keys for the pieces table
- Edits the foreign keys for the repertoire table
- Adds the routes for the student repertoire table
- Edits the routes for the pieces table
- Edits the routes for the repertoire table

### CODE REVIEW

**Comment by nathan-lalli** [comment]
There is a story for this on the storyboard

File: models/foreignkeys.js [hunk 1]
```diff
@@ -54,11 +55,8 @@ function addForeignKeys(db) {
   members.belongsTo(ensemble);
 
   pieces.belongsTo(composers);
-  pieces.belongsTo(repertoire);
+  pieces.belongsTo(studentrepertoire);
   composers.hasMany(pieces);
-  repertoire.hasMany(pieces);
-
-  repertoire.belongsTo(students);
```
Comment (ethanimooney) [inline models/foreignkeys.js:61]: These did not need to be removed - the piece still links to the repertoire, the studentrepertoire table only links the repertoire to the student.

Effectively like student -> studentinstructor <- instructor

student -> studentrepertoire <-repertoire

**Review by ethanimooney — CHANGES_REQUESTED** [review]

File: models/foreignkeys.js (hunk 1 above)
Comment (nathan-lalli) [reply models/foreignkeys.js]: Changed

**Review by ethanimooney — APPROVED** [review]
Looks good, thanks!

**Pull request merged**